winpkg/ddev_windows_installer.nsi

```diff
@@ -0,0 +1,1967 @@
+            Push "1"
+            Return
+        ${Else}
+            Push "Docker provider not accessible in WSL2: $R1"
+            Call LogPrint
+            Push "0"
+            Return
+        ${EndIf}
+    ${Else}
+        ; Check if Docker is accessible on Windows (traditional or WSL2 Docker CE setup)
+        Push "Checking Docker provider on Windows..."
+        Call LogPrint
+        nsExec::ExecToStack 'docker ps'
+        Pop $R0
+        Pop $R1
+        ${If} $R0 == 0
+            Push "Docker provider found on Windows: $R1"
+            Call LogPrint
+            Push "1"
+            Return
+        ${Else}
+            Push "Docker provider not accessible on Windows: $R1"
+            Call LogPrint
+            Push "0"
+            Return
+        ${EndIf}
+    ${EndIf}
+FunctionEnd
+
+Function DockerCheckPage
+    Push "Starting DockerCheckPage..."
+    Call LogPrint
+    
+    ; Skip this page for wsl2-docker-ce since Docker CE will be installed during the process
+    ${If} $INSTALL_OPTION == "wsl2-docker-ce"
+        Push "Skipping Docker check for wsl2-docker-ce install (Docker CE will be installed)"
+        Call LogPrint
+        Abort
+    ${EndIf}
+    
+    ; Skip this page if install type was specified via command line
+    ${If} $SILENT_INSTALL_TYPE != ""
+        Push "Skipping Docker check page for command line install"
+        Call LogPrint
+        Abort
+    ${EndIf}
+
+    ; Check for Docker provider
+    Push "Checking for Docker provider before proceeding..."
+    Call LogPrint
+    Call CheckDockerProvider
+    Pop $R0
+    ${If} $R0 == "1"
+        Push "Docker provider found, proceeding with installation"
+        Call LogPrint
+        Abort ; Skip this page since Docker is already working
+    ${EndIf}
+
+    ; Docker not found - show page to inform user
+    Push "Docker provider not found, showing information page"
+    Call LogPrint
+    nsDialogs::Create 1018
+    Pop $0
+    ${If} $0 == error
+        Push "Failed to create Docker check dialog"
+        Call LogPrint
+        Abort
+    ${EndIf}
+
+    ; Create different messages based on installation type
+    ${If} $INSTALL_OPTION == "traditional"
+        ${NSD_CreateLabel} 0 0 100% 60u "Docker provider is required for traditional Windows installation but was not found or is not running.$\r$\n$\r$\nPlease install and start Docker Desktop (https://www.docker.com/products/docker-desktop/) or Rancher Desktop (https://rancherdesktop.io/) before installing DDEV.$\r$\n$\r$\nYou can exit now to install Docker, or cancel this installation."
+    ${ElseIf} $INSTALL_OPTION == "wsl2-docker-desktop"
+        ${NSD_CreateLabel} 0 0 100% 60u "Docker Desktop or Rancher Desktop is required for this installation but is not accessible in WSL2.$\r$\n$\r$\nPlease install Docker Desktop or Rancher Desktop and enable WSL2 integration for the '$SELECTED_DISTRO' distro.$\r$\n$\r$\nYou can exit now to configure Docker, or cancel this installation."
```

Comment:
The installer doesn't say that Docker Desktop or Rancher Desktop should be running:

```suggestion
        ${NSD_CreateLabel} 0 0 100% 60u "Docker Desktop or Rancher Desktop is required for this installation but is not accessible in WSL2.$\r$\n$\r$\nEnsure that Docker Desktop or Rancher Desktop is installed, running, and has WSL2 integration enabled for the '$SELECTED_DISTRO' distro.$\r$\n$\r$\nThe 'docker ps' command must succeed inside WSL2 before launching this installer.$\r$\n$\r$\nYou can exit now to configure Docker, or cancel this installation."
```